src/wp-includes/blocks.php

```diff
@@ -157,23 +157,68 @@ function register_block_style_handle( $metadata, $field_name ) {
 	if ( empty( $metadata[ $field_name ] ) ) {
 		return false;
 	}
+
+	// Check whether styles should have a ".min" suffix or not.
+	$suffix = SCRIPT_DEBUG ? '' : '.min';
+
+	// Check whether this is a core block or not.
+	$is_core_block = isset( $metadata['name'] ) && 0 === strpos( $metadata['name'], 'core/' );
```

Comment:
I don't think it is the most reliable way to use it for handling core blocks. I would personally feel more comfortable if the check for style handle would remain unchanged and we would only tweak the part where `plugins_url` is used. Based on the location of the `block.json` we can detect whether it's inside the plugin, theme, or WordPress core and use the appropriate function to create `$style_uri`.

Ideally, we update `block.json` for all files to start using the relative path to CSS files so it resolves correctly depending on if it's being served from the Gutenberg plugin or WordPress core. The only challenge I see that in WordPress 5.6 and 5.7 we would have to polyfill in the Gutenberg plugin the part that uses `wp_style_add_data` to set `path`.

As to move forward with this patch we could add for 2 weeks the logic that registers styles that use the same handles as `block.json` holds.

Reply:
> I don't think it is the most reliable way to use it for handling core blocks. I would personally feel more comfortable if the check for style handle would remain unchanged and we would only tweak the part where `plugins_url` is used. Based on the location of the `block.json` we can detect whether it's inside the plugin, theme, or WordPress core and use the appropriate function to create `$style_uri`.

Refactored that part a bit, I believe it's simpler now. It now detects whether it's core or in a plugin using the JSON file's path. It doesn't detect whether it's in a theme, because themes are not allowed to register blocks.